mkosi/initrd/__main__.py

```diff
@@ -0,0 +1,102 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import argparse
+import logging
+import platform
+import shutil
+import tempfile
+
+from pathlib import Path
+from mkosi.config import OutputFormat, __version__
+from mkosi.log import log_setup
+from mkosi.run import find_binary, run, uncaught_exception_handler
+from mkosi.types import PathString
+
+
+@uncaught_exception_handler()
+def main() -> None:
+    log_setup()
+
+    parser = argparse.ArgumentParser(
+        prog="mkosi-initrd",
+        description="Build initrds or Unified Kernel Images using mkosi",
+        allow_abbrev=False,
+        usage="mkosi-initrd [options...]",
+    )
+
+    parser.add_argument("--kernel-version",
+                        metavar="KERNEL_VERSION",
+                        help="Kernel version string",
+                        default=platform.uname().release)
+    parser.add_argument("--format",
+                        choices=["cpio", "uki"],
+                        help="Output format (CPIO archive or UKI)",
+                        default="cpio")
+    parser.add_argument("--version",
+                        action="version",
+                        version=f"mkosi {__version__}")
+
+    args_initrd, args_other = parser.parse_known_args()
+
+    cmdline: list[PathString] = [
+        "mkosi",
+        "--directory", "",
+        "--format", str(args_initrd.format),
+        "--workspace-dir=/var/tmp",
+        "--package-cache-dir=/var",
+        "--cache-only=metadata",
+        "--extra-tree", f"/usr/lib/modules/{args_initrd.kernel_version}:/usr/lib/modules/{args_initrd.kernel_version}",
+        "--extra-tree=/usr/lib/firmware:/usr/lib/firmware",
+        "--remove-files=/usr/lib/firmware/*-ucode",
+        "--kernel-modules-exclude=.*",
+        "--kernel-modules-include=host",
+        "--include=mkosi-initrd",
+    ]
+
+    for d in ("/usr/lib/mkosi-initrd", "/etc/mkosi-initrd"):
+        if Path(d).exists():
+            cmdline += ["--include", d]
+
+    with tempfile.TemporaryDirectory() as d:
+        # Make sure we don't use any of mkosi's default repositories.
+        for p in (
+            "yum.repos.d/mkosi.repo",
+            "apt/sources.list",
```

Comment:
I updated this in 70aa901697f12182ccaa24e2325867d275479b55 which causes a merge conflict so this will need to be updated to match that commit